langchain/llms/base.py

```diff
@@ -34,6 +55,38 @@ def _get_verbosity() -> bool:
     return langchain.verbose
 
 
+def create_base_retry_decorator(
+    error_types: List[Type[BaseException]], max_retries: int = 1
+) -> Callable[[Any], Any]:
+    """Create a retry decorator for a given LLM and provided list of error types."""
+    import logging
+
+    logger = logging.getLogger(__name__)
+
+    min_seconds = 4
+    max_seconds = 10
+    # Wait 2^x * 1 second between each retry starting with
+    # 4 seconds, then up to 10 seconds, then 10 seconds afterwards
+    if len(error_types) == 0:
+        return lambda x: x
+    elif len(error_types) == 1:
+        retries = retry_if_exception_type(error_types[0])
+    else:
+        init = retry_if_exception_type(error_types[0])
+        retries = functools.reduce(
+            lambda x, y: x | retry_if_exception_type(y),  # type: ignore
```

Comment:
is there a simpler way to do this?

Reply:
Please, take a look, maybe it's more readable without `functools`.